server/src/main/java/org/elasticsearch/index/engine/CombinedDeletionPolicy.java

```diff
@@ -134,7 +138,8 @@ public void onCommit(List<? extends IndexCommit> commits) throws IOException {
         assert assertSafeCommitUnchanged(safeCommit);
         if (commitsListener != null) {
             if (newCommit != null) {
-                commitsListener.onNewAcquiredCommit(newCommit);
+                final Set<String> additionalFiles = listOfNewFileNames(previousLastCommit, newCommit);
```

Comment:
According to what I read from your comment, these commits are in order (one after the other, rather than mixed).